docs/certificate-specification.md

```diff
@@ -0,0 +1,112 @@
+# Certificate Specification
+
```

Comment:
Tiny nitpick: might want to reference https://www.ietf.org/rfc/rfc2119.txt for the semantics of `MUST`, `SHOULD`, etc.